controller/contour/deployment_test.go

```diff
@@ -0,0 +1,142 @@
+package contour
+
+import (
+	"testing"
+
+	operatorv1alpha1 "github.com/projectcontour/contour-operator/api/v1alpha1"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TODO [danehans]: Add integration tests to test reconcilitation
+// after https://github.com/projectcontour/contour-operator/pull/29 merges.
+
+var (
+	testName = "test-deploy"
+	testNs   = testName + "-ns"
+)
+
+func checkDeploymentHasEnvVar(t *testing.T, deploy *appsv1.Deployment, name string) {
+	t.Helper()
+
+	for _, envVar := range deploy.Spec.Template.Spec.Containers[0].Env {
+		if envVar.Name == name {
+			break
+		}
+	}
```

Comment:
The function doesn't return a value or raise an error.
```suggestion
			return
		}
	}
	t.Errorf("deployment is missing environment variable %q", name)
```